datatypes/src/primitives/multi_point.rs

```diff
@@ -285,6 +286,18 @@ where
     }
 }
 
+impl ApproxEq for MultiPoint {
+    type Margin = F64Margin;
+
+    fn approx_eq<M: Into<Self::Margin>>(self, other: Self, margin: M) -> bool {
+        let m = margin.into();
+        self.coordinates
+            .iter()
+            .zip(other.coordinates.iter())
+            .all(|(&a, &b)| a.approx_eq(b, m))
+    }
+}
```

Comment:
Does your implementation work when the `MultiPoint`s have differently sized lengths?

Please add a test for that and re-check your other implementations.

Consider one `MultiPoint` with length 3 and one with length 4 and the first three values are equal.

Reply:
598074c